### ISSUE TITLE
The deps.edn profile needs to be specified for the poly REPL to work with emacs/cider

### ISSUE THREAD

**Issue by OliverM** [polyfy/polylith#306]
I struggled a bit trying to launch a REPL in a poly workspace using emacs/cider's `cider-jack-in` command and then trying to `(require ...) a library interactively (or specify one in the `ns` declaration). I'd get an error like the following:

```
Execution error (FileNotFoundException) at dev.furkan/eval7127$loading (REPL:43).
Could not locate portal/api__init.class, portal/api.clj or portal/api.cljc on classpath.
```

If I started a repl in the project root using the dev and test aliases via `clj -A:dev:test` I could include libraries successfully.

It turns out Emacs/cider needs to be configured to use the dev & test profiles to allow libraries to be found on the class path. The [Practicalli Spaceman's guide](https://practical.li/spacemacs/clojure-development/project-configuration/) shows how.

If I'm missing something obvious that means this step isn't needed sorry about the noise! But I couldn't see how to make it work other than by the config above.

**Comment by furkan3ayraktar** [comment]
This is not specific to Emacs/Cider. Polylith utilizes `:dev` and `:test` aliases to provide a single development experience that includes all your bricks in a single REPL. Hence, if you do not start your REPL with those aliases, the classpath will not contain any of your bricks. 

You can check out the following for more information about this:
- https://polylith.gitbook.io/poly/architecture/development
- https://polylith.gitbook.io/poly/workflow/profile

Realworld example mentions about this [here]( https://github.com/furkan3ayraktar/clojure-polylith-realworld-example-app#start-it-in-your-clojure-repl) as well.

**Comment by OliverM** [comment]
Neither of those pages (which I'd read when searching for a solution before opening the issue) helped me. I've used emacs/cider with many other Clojure projects using `deps.edn` before without having to set the config I linked to above. The key was having to add
```
((clojure-mode . ((cider-clojure-cli-global-options . "-A:dev:test"))))
``` 
to a `.dir-locals.el` file in the project root directory. Given that I've not had to do this for other Clojure projects I thought it might be useful for other emacs/cider users to have this in a footnote in the docs somewhere, and I also thought you might consider including that given the extensive (and welcome) guidance here for Cursive users.
